utils_test.go

```diff
@@ -42,3 +50,202 @@ func TestValidStoragePathFormat(t *testing.T) {
 		assert.NilError(t, err)
 	}
 }
+
+func TestRootlessRuntimeDir(t *testing.T) {
```

Comment:
(Non-blocking? This feels long and laborious and a bit hard to follow; it could probably be shorter and clearer as a Go-style table-driven test.)

Reply:
I am not very experienced in `go`. What table tests do you mean?

Reply:
https://github.com/golang/go/wiki/TableDrivenTests : have shared code that performs the test, with the different situations described as just data.

Reply:
Aaa, no. This test can't be implemented in that way: it is branched and next branch depends on previous. We can make it smaller and more readable but can't make it table like.

Reply:
Do the individual cases really _strongly_ depend on the sequence they happen?
A
```go
[]struct {
   runtimeDirPresent bool
   proc1CommReadable bool
   proc1Comm string
   runUserUIDDirPresent bool
   runUserUIDDirPermissions int
   tmpUIDDirPresent bool
   tmpUIDDirPermissions int
   expectedPath string // "" on expected error
}
```
or something like that, seems possible. Or maybe I misunderstand what “it is branched” refers to.

Reply:
It is possible but we will receive branched table: first item will have only one `runtimeDirPresent` filled, last item will have all filled. I can implement it if you like, no problem.

Reply:
As mentioned at the start, this is non-blocking for me. Looking at all the members above, it’s not even obvious that it would be shorter … but it should be easier to follow the logic.

(BTW filling all of the fields with valid possibilities along with `runtimeDirPresent` and testing that the `runtimeDir` is the one chosen would also be a reasonable test — and much easier to add in the table-driven implementation. But then, it’s not a security-relevant test so I’m not asking for that to be added.)